edge_detection/params.py

```diff
@@ -0,0 +1,129 @@
+#  Copyright (c) 2024 Mira Geoscience Ltd.
+#
+#  This file is part of geoapps.
+#
+#  geoapps is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Optional, Union
+
+from geoh5py.data import FloatData
+from geoh5py.objects import Grid2D
+from geoh5py.ui_json import InputFile
+from geoh5py.workspace import Workspace
+from pydantic import BaseModel, ConfigDict
+
+
+class ApplicationParameters(BaseModel):
```

Comment:
Consider renaming this to DriverParameters.  We use 'application' in geoapps to denote the notebook/dash rendering,

Reply:
..Or EdgeDetectionParameters?  Maybe even just Parameters since there's no need to distinguish anymore within a class hierarchy?